crates/fuel-core/src/query/balance.rs

```diff
@@ -28,29 +29,28 @@ use std::{
 pub mod asset_query;
 
 impl ReadView {
-    pub fn balance(
+    pub async fn balance(
         &self,
         owner: Address,
         asset_id: AssetId,
         base_asset_id: AssetId,
     ) -> StorageResult<AddressBalance> {
         let amount = AssetQuery::new(
             &owner,
-            &AssetSpendTarget::new(asset_id, u64::MAX, usize::MAX),
+            &AssetSpendTarget::new(asset_id, u64::MAX, u16::MAX),
             &base_asset_id,
             None,
             self,
         )
         .coins()
         .map(|res| res.map(|coins| coins.amount()))
-        .try_fold(0u64, |mut balance, res| -> StorageResult<_> {
-            let amount = res?;
-
-            // Increase the balance
-            balance = balance.saturating_add(amount);
-
-            Ok(balance)
-        })?;
+        .try_fold(0u64, |balance, amount| {
+            async move {
+                // Increase the balance
+                Ok(balance.saturating_add(amount))
```

Comment:
this behaviour would be incorrect if the user has a balance of > 18.45 trillion (possible for memecoin projects/?)

Reply:
but it's a major edge case so feel free to ignore

Reply:
It is how it worked before=)